dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/TrackerTest.java

```diff
@@ -274,6 +279,27 @@ public TrackedEntityInstance persistTrackedEntityInstanceWithEnrollmentAndEvents
         return _persistTrackedEntityInstanceWithEnrollmentAndEvents( 5, enrollmentValues );
     }
 
+    public void deleteOneEnrollment(
+        org.hisp.dhis.dxf2.events.trackedentity.TrackedEntityInstance trackedEntityInstance )
+    {
+        List<Enrollment> enrollments = trackedEntityInstance.getEnrollments();
+        if ( !enrollments.isEmpty() )
```

Comment:
Should these helpers have an assertion in them in case no enrollment/event was deleted?

When I write a test in the future using `deleteOneEnrollment()` I think I'd assume it did delete one.

It is probably also useful to return the deleted enrollment/event. So one can use that in assertions in tests to make sure some API does not return that deleted entity. With that you then also have the choice of not making any assertion in the helper but letting the test do it. Not sure which is the best one. I lean towards making the assertion in the helper.

Reply:
Done